system-helper/flatpak-system-helper.c

```diff
@@ -1808,6 +1808,31 @@ handle_update_summary (FlatpakSystemHelper   *object,
   return TRUE;
 }
 
+static gboolean
+handle_cancel_pull (FlatpakSystemHelper   *object,
+                    GDBusMethodInvocation *invocation,
+                    const gchar           *arg_installation,
+                    const gchar           *arg_repo_path)
+{
+  g_autoptr(FlatpakSystemHelperOngoingPull) ongoing_pull = NULL;
+  g_autoptr(FlatpakDir) system = NULL;
+  g_autoptr(GError) error = NULL;
+
+  g_debug ("CancelPull %s", arg_installation);
+
+  system = dir_get_system (arg_installation, &error);
+  if (system == NULL)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return TRUE;
+    }
+
+  ongoing_pull = get_ongoing_pull_by_dir (arg_repo_path);
```

Comment:
This is a little sneaky - the handling here is a bit too implicit for my taste. I would prefer it to not be just a side-effect of returning from this function

Reply:
hmm, I am unable to decipher this comment. Can you be a bit more verbose and suggest what you would like to see?

Reply:
My comment was that you don't do anything to handle the cancellation. It gets handled implicitly by assigning the pull to an autocleared variable, and leaving the scope. 
I would have expected to see you make a function call here, maybe suggestively named ongoing_pull_cancel (....)